cmd/frontend/graphqlbackend/graphqlutil/connection_resolver_test.go

```diff
@@ -0,0 +1,555 @@
+		t.Fatal(err)
+	}
+	if diff := cmp.Diff(2, len(nodes)); diff != "" {
+		t.Fatal(diff)
+	}
+
+	resolver.Nodes(ctx)
+	if diff := cmp.Diff(1, store.ComputeNodesCalled); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestConnectionNodesBeforeArgs(t *testing.T) {
+	store := &TestConnectionStore{t,
+		func(_ context.Context, args *database.PaginationArgs) {
+			expectedArgs := &database.PaginationArgs{
+				First:  toInt32(6),
+				Last:   nil,
+				After:  nil,
+				Before: toInt32(0),
+			}
+			if diff := cmp.Diff(expectedArgs, args); diff != "" {
+				t.Fatal(diff)
+			}
+		}, 0, 0}
+
+	beforeCursor := "0"
+	resolver, err := NewConnectionResolver[TestConnectionNode](store, &ConnectionResolverArgs{
+		First:  toInt32(5),
+		Last:   nil,
+		After:  nil,
+		Before: &beforeCursor,
+	},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx := context.Background()
+	nodes, err := resolver.Nodes(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if diff := cmp.Diff(2, len(nodes)); diff != "" {
+		t.Fatal(diff)
+	}
+
+	resolver.Nodes(ctx)
+	if diff := cmp.Diff(1, store.ComputeNodesCalled); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestConnectionNodesWithLimit(t *testing.T) {
+	store := &TestConnectionStore{t,
+		func(_ context.Context, args *database.PaginationArgs) {
+			expectedArgs := &database.PaginationArgs{
+				First:  toInt32(2),
+				Last:   nil,
+				After:  nil,
+				Before: toInt32(0),
+			}
+			if diff := cmp.Diff(expectedArgs, args); diff != "" {
+				t.Fatal(diff)
+			}
+		}, 0, 0}
+
+	beforeCursor := "0"
+	resolver, err := NewConnectionResolver[TestConnectionNode](store, &ConnectionResolverArgs{
+		First:  toInt32(1),
+		Last:   nil,
+		After:  nil,
+		Before: &beforeCursor,
+	},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx := context.Background()
+	nodes, err := resolver.Nodes(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if diff := cmp.Diff(1, len(nodes)); diff != "" {
+		t.Fatal(diff)
+	}
+
+	resolver.Nodes(ctx)
+	if diff := cmp.Diff(1, store.ComputeNodesCalled); diff != "" {
+		t.Fatal(diff)
+	}
+}
+
+func TestConnectionPageInfoDefault(t *testing.T) {
+	ctx := context.Background()
+	store := &TestConnectionStore{t, func(context.Context, *database.PaginationArgs) { return }, 0, 0}
+	resolver, err := NewConnectionResolver[TestConnectionNode](store, &ConnectionResolverArgs{First: toInt32(20), Last: nil, After: nil, Before: nil})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pageInfo, err := resolver.PageInfo(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	startCursor, err := pageInfo.StartCursor()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if diff := cmp.Diff("0", *startCursor); diff != "" {
+		t.Fatal(diff)
+	}
+
+	endCursor, err := pageInfo.EndCursor()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if diff := cmp.Diff("1", *endCursor); diff != "" {
+		t.Fatal(diff)
+	}
+
+	if diff := cmp.Diff(false, pageInfo.HasNextPage()); diff != "" {
+		t.Fatal(diff)
+	}
+
+	if diff := cmp.Diff(false, pageInfo.HasPreviousPage()); diff != "" {
+		t.Fatal(diff)
+	}
```

Comment:
It's booleans, we can assert those in a cleaner way :)

```suggestion
	if !pageInfo.HasNextPage() {
		t.Fatal("hasNextPage should be false, but is true")
	}

	if !pageInfo.HasPreviousPage() {
		t.Fatal("hasPreviousPage should be false, but is true")
	}
```